### ISSUE TITLE
CSharpGenerateVariableCodeFixProvider/GenerateMethodCodeFixProvider crash with `ArgumentException : Not a C# symbol.`

### ISSUE THREAD

**Issue by yyjdelete** [dotnet/roslyn#36814]
**Version Used**: 
Compiler version: '3.2.0-beta3-19312-05 (5905d757)'. Language version: default.
VS2019 16.2preview3(used to be work in 16.1.3)

**Steps to Reproduce**:
The same Exception message as #36702, but seems totally different stack. Not sure if it's related.

1. Type the below incompleted code
```cs
System.Runtime.CompilerServices.StrongBox<int> x = null;
//or any incompleted code(not exists field) here. Change int to var not crashes.
int val = System.Threading.Volatile.Read(ref x.Va);//Value
```

2. Put mouse or cursor near `x.Va`

**Expected Behavior**:
Not crash, just compile failed with some incompleted code.

**Actual Behavior**:
see CSharpGenerateVariableCodeFixProvider/GenerateMethodCodeFixProvider crash.

```cs
System.ArgumentException : Not a C# symbol.
Parameter name: typeArguments
   at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.EnsureCSharpSymbolOrNull[TSource,TDestination](TSource symbol,String paramName)
   at Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol.Microsoft.CodeAnalysis.IMethodSymbol.Construct(ITypeSymbol[] arguments)
   at Microsoft.CodeAnalysis.CSharp.CSharpTypeInferenceService.TypeInferrer.Instantiate(IMethodSymbol method,IList`1 invocationTypes)
   at Microsoft.CodeAnalysis.CSharp.CSharpTypeInferenceService.TypeInferrer.<>c__DisplayClass21_0.<InferTypeInArgument>b__2(IMethodSymbol m)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.CSharp.CSharpTypeInferenceService.TypeInferrer.InferTypeInArgument(Int32 index,IEnumerable`1 methods,ArgumentSyntax argumentOpt)
   at Microsoft.CodeAnalysis.CSharp.CSharpTypeInferenceService.TypeInferrer.InferTypeInInvocationExpression(InvocationExpressionSyntax invocation,Int32 index,ArgumentSyntax argumentOpt)
   at Microsoft.CodeAnalysis.CSharp.CSharpTypeInferenceService.TypeInferrer.InferTypeInArgument(ArgumentSyntax argument,Nullable`1 previousToken)
   at Microsoft.CodeAnalysis.CSharp.CSharpTypeInferenceService.TypeInferrer.InferTypesWorker_DoNotCallDirectly(SyntaxNode node)
   at Microsoft.CodeAnalysis.LanguageServices.TypeInferenceService.AbstractTypeInferenceService.AbstractTypeInferrer.InferTypes(SyntaxNode expression,Boolean filterUnusable)
   at Microsoft.CodeAnalysis.LanguageServices.TypeInferenceService.AbstractTypeInferenceService.InferTypes(SemanticModel semanticModel,SyntaxNode expression,String nameOpt,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.GenerateMember.GenerateVariable.AbstractGenerateVariableService`3.State.DetermineFieldType(SemanticDocument semanticDocument,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.GenerateMember.GenerateVariable.AbstractGenerateVariableService`3.State.TryInitializeSimpleName(TService service,SemanticDocument semanticDocument,TSimpleNameSyntax simpleName,CancellationToken cancellationToken)
   at async Microsoft.CodeAnalysis.GenerateMember.GenerateVariable.AbstractGenerateVariableService`3.State.TryInitializeAsync[TService,TSimpleNameSyntax,TExpressionSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.GenerateMember.GenerateVariable.AbstractGenerateVariableService`3.State.GenerateAsync[TService,TSimpleNameSyntax,TExpressionSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.GenerateMember.GenerateVariable.AbstractGenerateVariableService`3.GenerateVariableAsync[TService,TSimpleNameSyntax,TExpressionSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.GenerateMember.AbstractGenerateMemberCodeFixProvider.RegisterCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformActionAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by jcouv** [comment]
Tagging @jasonmalinowski @ryzngard

**Comment by jasonmalinowski** [comment]
This was fixed by #36517 and the fix will be in Preview 4. Thanks @yyjdelete for filing this though -- had we not had this already fixed we would have been jumping on it pronto.

**Comment by mavasani** [comment]
FYI: I have hit this couple of times today while typing. Will upgrade to latest Preview4 build and verify it does not repro anymore.
